xarray/core/datatree.py

```diff
@@ -510,6 +556,26 @@ def is_hollow(self) -> bool:
         """True if only leaf nodes contain data."""
         return not any(node.has_data for node in self.subtree if not node.is_leaf)
 
+    def _get_inherited_coord_var(self: DataTree, key: str) -> Variable:
+        for node in self.parents:
+            if key in node._local_coord_names:
+                return node._local_variables[key]
+
+        raise Exception(
+            "should never get here - means we didn't do our alignment / construction properly"
+        )
```

Comment:
This method, along with the various `._*` properties below, is what substitutes for all the `ChainMap` stuff in #9063.

Here we take the approach that we iterate up the tree to go fetch the inherited coordinate variable only when we need it. (As opposed to pre-fetching all of them and storing all of them on each child node under a `ChainMap`.)

Reply:
This function is basically a very literal way to do the "search by proximity" in the CF conventions.